src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1673,6 +1673,22 @@ void Lowering::ContainCheckBinary(GenTreeOp* node)
         }
     }
 
+    // Find "a - b * c" in order to emit MSUB
+    if (comp->opts.OptimizationEnabled() && varTypeIsIntegral(node) && !node->isContained() && node->OperIs(GT_SUB) &&
```

Comment:
May be, have some of the `node` related checks merged with the above condition?

```c++
if (comp->opts.OptimizationEnabled() && varTypeIsIntegral(node) && !node->isContained() && !node->gtOverflow())
{
   // Find "a * b + c" or "c + a * b" in order to emit MADD
  if ( .. )
  {
  }

   // Find "a - b * c" in order to emit MSUB
  if (..)
  {
  }
}
```

Reply:
Also, in the prior `if condition`, there is reference of `MSUB` instead of `MADD`. Can you please fix that as well?

Reply:
Combining them makes sense to me - also it depends on whether or not we want to use GT_MADD instead of introducing GT_MSUB.

Reply:
nit: `!node->gtOverflow()` can also be included in the common check.